#! /bin/sh -e

# DP: Fix bug in xml transformer, needed to fix eclipse's handling of its configuration files

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

2005-11-20  Chris Burdess  <dog@gnu.org>

        * gnu/xml/transform/TransformerImpl.java: Ensure that output stream
        is not closed if provided in the StreamResult.

Index: gnu/xml/transform/TransformerImpl.java
===================================================================
RCS file: /cvsroot/classpath/classpath/gnu/xml/transform/TransformerImpl.java,v
retrieving revision 1.6
diff -u -r1.6 TransformerImpl.java
--- libjava/gnu/xml/transform/TransformerImpl.java      30 Sep 2005 07:17:04 -0000      1.6
+++ libjava/gnu/xml/transform/TransformerImpl.java      20 Nov 2005 12:02:48 -0000
@@ -487,6 +487,7 @@
     throws IOException
   {
     OutputStream out = null;
+    boolean created = false;
     try
       {
         out = sr.getOutputStream();
@@ -523,6 +524,7 @@
                 URL url = new URL(systemId);
                 out = new FileOutputStream(url.getPath());
               }
+            created = true;
           }
         out = new BufferedOutputStream(out);
         StreamSerializer serializer =
@@ -539,7 +541,7 @@
       {
         try
           {
-            if (out != null)
+            if (out != null && created)
               {
                 out.close();
               }
